src/node.cc

```diff
@@ -3682,9 +3682,10 @@ static void ParseArgs(int* argc,
       force_repl = true;
     } else if (strcmp(arg, "--no-deprecation") == 0) {
       no_deprecation = true;
-    } else if (strcmp(arg, "--napi-modules") == 0) {
-      load_napi_modules = true;
-    } else if (strcmp(arg, "--napi-modules=yes") == 0) {
+    } else if (strcmp(arg, "--napi-modules") == 0 ||
+               strcmp(arg, "--napi-modules=yes") == 0) {
+      fprintf(stderr, "Warning: N-API is an experimental feature "
+        "and could change at any time.\n");
```

Comment:
Oh I see there is a `ProcessEmitWarning()` function in `node.cc`. I'd just need to find a good place to call it... command-line parsing is too early.